vm/array_test.go

```diff
@@ -53,6 +53,38 @@ func TestPushMethod(t *testing.T) {
 	}
 }
 
+func TestShiftMethod(t *testing.T) {
```

Comment:
I think we can separate functional tests and error tests into two different test methods.